apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/internal/RealCacheKeyBuilder.kt

```diff
@@ -29,34 +29,39 @@ class RealCacheKeyBuilder : CacheKeyBuilder {
   }
 
   @Suppress("UNCHECKED_CAST")
-  private fun resolveArguments(objectMap: Map<String, Any?>, variables: Operation.Variables): Map<String, Any?> {
-    return objectMap.mapValues { (_, value) ->
-      if (value is Map<*, *>) {
-        val nestedObjectMap = value as Map<String, Any?>
-        if (isArgumentValueVariableType(nestedObjectMap)) {
-          resolveVariableArgument(nestedObjectMap, variables)
+  private fun resolveVariables(value: Any?, variables: Operation.Variables): Any? {
+    return when (value) {
+      null -> null
+      is Map<*, *> -> {
+        value as Map<String, Any?>
+        if (isArgumentValueVariableType(value)) {
+          resolveVariable(value, variables)
         } else {
-          resolveArguments(nestedObjectMap, variables)
+          value.mapValues {
+            resolveVariables(it.value, variables)
+          }.toList()
+              .sortedBy { it.first }
```

Comment:
toSortedMap() fits also
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-sorted-map.html

Reply:
I wish it worked but unfortunately `toSortedMap()` is JVM only. At this point this is mainly rhetorical because there's no real way to consume the normalized cache from anything else than JVM but there will be a native cache in the future.